config/v1/types_authentication.go

```diff
@@ -271,6 +296,99 @@ type TokenClaimMapping struct {
 	Claim string `json:"claim"`
 }
 
+// +kubebuilder:validation:XValidation:rule="!(has(self.claim) && has(self.expression))",message="both claim and expression must not be set"
+// +kubebuilder:validation:XValidation:rule="has(self.claim) || has(self.expression)",message="either claim or expression must be set"
+type TokenClaimOrExpressionMapping struct {
+	// claim is an optional field for specifying the
+	// JWT token claim that is used in the mapping.
+	// The value of this claim will be assigned to
+	// the field in which this mapping is associated.
+	//
+	// Either claim or expression must be set.
+	// claim must not be specified when expression is set.
+	//
+	// +optional
+	Claim string `json:"claim"`
```

Comment:
Are there any validations that can be added to limit the set of characters that may be used within this string?

Reply:
I did look for a general pattern of JWT claims, but couldn't find anything that seemed reasonable. For the most part, it looks like lowercase alpha with hyphens and underscores would be OK but some OIDC providers seem to allow setting totally custom claims.

I think it makes more sense to let this be free form. There is no format or length validation on the Kubernetes API server side for this field either.